Dhcp option check mode

willthames opened the pull request:
##### SUMMARY
ec2_vpc_dhcp_options_facts supports check mode by default.

Also various tidy ups based on Ansible and AWS module best practices (in a separate commit)

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
ec2_vpc_dhcp_options_facts

##### ANSIBLE VERSION
```
ansible 2.4.0 (devel c1b3d6a51f) last updated 2017/03/30 09:17:50 (GMT +1000)
  config file = ./ansible.cfg
  configured module search path = [u'./library']
  python version = 2.7.13 (default, Jan 12 2017, 17:59:37) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]
```


##### ADDITIONAL INFORMATION

Commit:
Support check mode in ec2_vpc_dhcp_options_facts

As a facts module, ec2_vpc_dhcp_options_facts supports check mode
by default

ryansb commented on lib/ansible/modules/cloud/amazon/ec2_vpc_dhcp_options_facts.py:
```diff
@@ -34,12 +34,13 @@
       - Provide this value as a list
     required: false
     default: None
+    aliases: ['DhcpOptionsIds']
```
:+1:

ryansb commented on lib/ansible/modules/cloud/amazon/ec2_vpc_dhcp_options_facts.py:
```diff
@@ -83,91 +84,75 @@
+    params = dict(Filters=ansible_dict_to_boto3_filter_list(module.params.get('filters')))
+
+    if module.params.get("dry_run"):
+        params['DryRun'] = module.params.get("dry_run")
```
Shouldn't this just be `params['DryRun'] = True` since the dry_run property is a boolean, so it'll only hit this when it's true anyway?

willthames replied:
Fixed

ryansb reviewed (changes requested):
Small tweak, otherwise looks great.

Commit:
ec2_vpc_dhcp_options_facts tidy up

Use named method imports, move imports to top of code
Use shared code to handle filters and tags
Use snake case for parameter names while retaining backward compatibility
